src/Discord.Net.Core/Entities/Interactions/Modals/ModalBuilder.cs

```diff
@@ -108,21 +108,100 @@ public ModalBuilder AddComponents(List<IMessageComponent> components, int row)
             return this;
         }
 
+        /// <summary>
+        ///     Gets a <see cref="TMessageComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <typeparam name="TMessageComponent">The type of the component to get.</typeparam>
+        /// <param name="customId">The <see cref="IMessageComponent.CustomId"/> of the component to get.</param>
+        /// <returns>The component of type <typeparamref name="TMessageComponent"/> that was found.</returns>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when the <typeparamref name="TMessageComponent"/> to be updated was not found.
+        /// </exception>
+        public TMessageComponent GetComponent<TMessageComponent>(string customId) 
+            where TMessageComponent : class, IMessageComponent 
+            => Components.ActionRows?.SelectMany(r => r.Components.OfType<TMessageComponent>()).FirstOrDefault(c => c?.CustomId == customId)
+                ?? throw new ArgumentException($"There is no component of type {typeof(TMessageComponent).Name} with the specified custom ID in this modal builder.", nameof(customId));
+
+        /// <summary>
+        ///     Updates a <see cref="TextInputComponent"/> by the specified <paramref name="customId"/>.
+        /// </summary>
+        /// <param name="customId">The <see cref="TextInputComponent.CustomId"/> of the input to update.</param>
+        /// <param name="updateTextInput">An action that configures the updated text input.</param>
+        /// <returns>The current builder.</returns>
+        public ModalBuilder UpdateTextInput(string customId, Action<TextInputBuilder> updateTextInput)
+        {
+            var component = GetComponent<TextInputComponent>(customId);
```

Comment:
possible NRE

Reply:
would this method really be redundant? there are many who create such extensions for this

Reply:
??
there's a possible `NullReferenceException` here if the `customId` is null

Reply:
sorry, misunderstood you, in this case, should I use `Preconditions.NotNull(customId, nameof(customId)`?

Reply:
uh, misunderstood you